chains/solana/transactions/addAgreement.js

```diff
@@ -0,0 +1,40 @@
+import * as anchor from "@coral-xyz/anchor";
+import { PublicKey } from "@solana/web3.js";
+
+export const addAgreement = async (details) => {
+  const companyAccount = details.publicKey
+
+  const agreementData = {
+    title: details.title,
+    description: details.description,
+    skills: details.skills,
+    professional: new PublicKey(details.professional),
+    company: companyAccount,
+  }
+
+  const stringBuffer = Buffer.from("company_agreements", "utf-8");
+  const [companyAgreementsPublicKey, _] = anchor.web3.PublicKey.findProgramAddressSync(
+    [stringBuffer, companyAccount.toBuffer()],
+    details.contract.programId
+  );
+
+  try {
+    const agreementAddress = anchor.web3.Keypair.generate();
+
+    const tx = await details.contract.methods
+      .initializeAgreement(agreementData).accounts({
+          agreement: agreementAddress.publicKey,
```

Comment:
there is an important info missing here: which community this professional / agreement belongs

Reply:
Noted, will add this feature to the Solana program.